aloha-core/src/main/scala/com/eharmony/aloha/models/multilabel/MultilabelModel.scala

```diff
@@ -0,0 +1,387 @@
+import spray.json.{JsonFormat, JsonReader}
+
+import scala.collection.{immutable => sci, mutable => scm}
+import scala.util.{Failure, Success, Try}
+
+
+// TODO: When adding label-dep features, a Seq[GenAggFunc[K, Sparse]] will be needed.
+// TODO: To create a Seq[GenAggFunc[K, Sparse]], a Semantics[K] needs to be derived from a Semantics[A].
+// TODO: MorphableSemantics provides this.  If K is *embedded inside* A, it should be possible in some cases.
+// TODO: An alternative is to pass a Map[K, Sparse], Map[K, Option[Sparse]], Map[K, Seq[Sparse]] or something.
+// TODO: Directly passing the map of LDFs avoids the need to derive a Semantics[K].  This is easier to code.
+// TODO: Directly passing LDFs would however be more burdensome to the data scientists.
+
+/**
+  * A multi-label predictor.
+  *
+  * Created by ryan.deak on 8/29/17.
+  *
+  * @param modelId An identifier for the model.  Used in score and error reporting.
+  * @param featureNames feature names (parallel to featureFunctions)
+  * @param featureFunctions feature extracting functions.
+  * @param labelsInTrainingSet a sequence of all labels encountered during training. Note: the
+  *                            order of labels may relate to the predictor produced by
+  *                            predictorProducer.  It is the caller's responsibility to ensure
+  *                            the order is correct.  To mitigate such problems, both labels
+  *                            and indices into labelsInTrainingSet are passed to the predictor
+  *                            produced by predictorProducer.
+  * @param labelsOfInterest if provided, a sequence of labels will be extracted from the example
+  *                         for which a prediction is desired. The ''intersection'' of the
+  *                         extracted labels and the training labels will be the labels for which
+  *                         predictions will be produced.
+  * @param predictorProducer the function produced when calling this function is responsible for
+  *                          getting the data into the correct type and using it within an
+  *                          underlying ML library to produce a prediction.  The mapping back to
+  *                          (K, Double) pairs is also its responsibility.  If the predictor
+  *                          produced by predictorProducer is Closeable, it will be closed when
+  *                          MultilabelModel's close method is called.
+  * @param numMissingThreshold if provided, we check whether the threshold is exceeded.  If so,
+  *                            return an error instead of the computed score.  This is for missing
+  *                            data situations.
+  * @param auditor transforms a `Map[K, Double]` to a `B`.  Reports successes and errors.
+  * @tparam U upper bound on model output type `B`
+  * @tparam K type of label or class
```

Comment:
`K` is not constrained.  Should we make the constructor private and then have two factory methods: one where `K <: AnyVal` and one where `K <: Serializable`?  The could both thunk down to the unconstrained constructor.  What do you think about this?  If `K` is not constrained, we can **guarantee** the serializability of `MultilabelModel`.

Reply:
Yes, for sure we should do this.  This class has to be serializable.  If that's not a test, please add it.

Reply:
I added a type class that can ensure types are `Serializable` or `AnyVal`.  I like this approach better than making the constructor private and adding multiple factory methods.